src/app/chat/chat-list/chat-list.component.ts

```diff
@@ -172,50 +175,28 @@ export class ChatListComponent {
    * Will show a popup confomation.
    * if user conform, then it call _createCohortChannelHandler method to create cohort channel.
    */
-  createCohortChatChannel() {
-    this.notification.alert({
-      cssClass: 'chat-conformation',
+  async createCohortChatChannel() {
+    const modal = await this.modalController.create({
+      component: GroupChatPopupComponent,
+      cssClass: 'chat-group-message-popup',
+      keyboardClose: false,
       backdropDismiss: false,
-      subHeader: 'Create cohort channel?',
-      message: 'Are you sure you want to create cohort channel?',
-      buttons: [
-        {
-          text: 'Create',
-          handler: () => {
-            this._createCohortChannelHandler();
-          }
-        },
-        {
-          text: 'Cancel',
-          role: 'cancel'
-        }
-      ]
-    });
-  }
-
-  /**
-   * Call chat service to create cohort channel
-   */
-  private _createCohortChannelHandler() {
-    const timeLineId = this.storage.getUser().timelineId;
-    const timelineUuid = this.storage.getUser().timelineUuid;
-    const currentProgram = this.storage.get('programs').find(program => {
-      return program.timeline.id === timeLineId;
+      componentProps: {
+        createdChannels: this.checkChannelsCreated('cohort'),
+      }
     });
-    this.chatService.createChannel({
-      name: currentProgram.timeline.title,
-      isAnnouncement: false,
-      roles: ['participant', 'mentor', 'admin', 'coordinator'],
-      members: [{
-        type: 'Timeline',
-        uuid: timelineUuid
-      }]
-    }).subscribe(chat => {
-      if (!this._channelExist(chat, 'cohort')) {
-        this.chatChannels.push(chat);
-        this._groupingChatChannels();
+    await modal.present();
+    modal.onWillDismiss().then((data) => {
+      if (data.data && data.data.newChannel) {
+        if (!this._channelExist(data.data.newChannel, 'cohort')) {
+          this.chatChannels.push(data.data.newChannel);
+          this._groupingChatChannels();
+          // Subscribe to the pusher channel of new create chat channel.
+          this._checkAndSubscribePusherChannels();
+          this.goToChatRoom(data.data.newChannel);
```

Comment:
Overall, don't do this kind of modifications after we create a new channel. It's so easy to make mistake.
Instead, we can just retrieve the chat list again from API.